### PR TITLE
Player projectile

### PR DESCRIPTION
**Pull request by Jodiiiiiiii**
mouse inputs used to spawn projectiles that bounce with no loss of energy. 

ability on 5 second cooldown. projectiles expire after 5 seconds.

Note: I made this branch as a branch off of the ColliderInconsistencies-Bug branch, so the changes from that show up here - sorry, not sure if there is a better way to handle multiple pull requests at once but I figured they each deserved their own branch
-    TL;DR merge "Collider inconsistencies bug" PR first

### CODE REVIEW

**Review by Andrew-Kohler — APPROVED** [review]
I would usually recommend handling the projectile ability in a separate class, mostly because continuing to add player abilities to the controller is going to get messy/long. I don't think this is going to get long enough for it to become an actual issue, so I'm good to merge it in for sake of time. I'll leave both of these open until Saturday and then merge them in if nobody else objects.
